File: CS/Q101293/Form1.cs
using System;
using System.Data;
using System.Windows.Forms;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Filtering;
using DevExpress.Data.Filtering.Helpers;
using DevExpress.Utils.Menu;

namespace Q101293 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("Check", typeof(bool));
            for(int i = 1; i < 10; i++)
                tbl.Rows.Add(i, string.Format("Item {0}", i), i % 2 == 0);
            gridControl1.DataSource = tbl;
            gridView1.ActiveFilterString = "[Check] = false";
        }

        private void gridView1_FilterEditorCreated(object sender, DevExpress.XtraGrid.Views.Base.FilterControlEventArgs e) {
            e.FilterControl.PopupMenuShowing += OnFilterControlPopupMenuShowing;
        }

        void OnFilterControlPopupMenuShowing(object sender, PopupMenuShowingEventArgs e) {
            if (e.MenuType == FilterControlMenuType.Group) {
                // Hide commands
                e.Menu.Remove(GroupType.NotAnd);
                e.Menu.Remove(GroupType.NotOr);
                //Locate and then disable and rename commands.
                e.Menu.Find(StringId.FilterMenuGroupAdd).Enabled = false;
                e.Menu.Find(StringId.FilterMenuClearAll).Caption = "Remove All";
            }
            // Hide all operators except Equals and DoesNotEqual for the "ID" field
            if (e.MenuType == FilterControlMenuType.Clause) {
                ClauseNode node = e.CurrentNode as ClauseNode;
                if (node.Property.Name == "ID") {
                    DXMenuItem itemEqual = e.Menu.Find(ClauseType.Equals);
                    DXMenuItem itemNotEqual = e.Menu.Find(ClauseType.DoesNotEqual);
                    for (int i = e.Menu.Items.Count - 1; i >= 0; i--) {
                        DXMenuItem item = e.Menu.Items[i];
                        if (!item.Equals(itemEqual) && !item.Equals(itemNotEqual))
                            item.Visible = false;
                    }
                }
            }
        }
    }
}

File: VB/Q101293/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Filtering
Imports DevExpress.Data.Filtering.Helpers
Imports DevExpress.Utils.Menu

Namespace Q101293
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Dim tbl As New DataTable()
            tbl.Columns.Add("ID", GetType(Integer))
            tbl.Columns.Add("Name", GetType(String))
            tbl.Columns.Add("Check", GetType(Boolean))
            For i As Integer = 1 To 9
                tbl.Rows.Add(i, String.Format("Item {0}", i), i Mod 2 = 0)
            Next i
            gridControl1.DataSource = tbl
            gridView1.ActiveFilterString = "[Check] = false"
        End Sub

        Private Sub gridView1_FilterEditorCreated(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.FilterControlEventArgs) Handles gridView1.FilterEditorCreated
            AddHandler e.FilterControl.PopupMenuShowing, AddressOf OnFilterControlPopupMenuShowing
        End Sub

        Private Sub OnFilterControlPopupMenuShowing(ByVal sender As Object, ByVal e As PopupMenuShowingEventArgs)
            If e.MenuType = FilterControlMenuType.Group Then
                ' Hide commands
                e.Menu.Remove(GroupType.NotAnd)
                e.Menu.Remove(GroupType.NotOr)
                'Locate and then disable and rename commands.
                e.Menu.Find(StringId.FilterMenuGroupAdd).Enabled = False
                e.Menu.Find(StringId.FilterMenuClearAll).Caption = "Remove All"
            End If
            ' Hide all operators except Equals and DoesNotEqual for the "ID" field
            If e.MenuType = FilterControlMenuType.Clause Then
                Dim node As ClauseNode = TryCast(e.CurrentNode, ClauseNode)
                If node.Property.Name = "ID" Then
                    Dim itemEqual As DXMenuItem = e.Menu.Find(ClauseType.Equals)
                    Dim itemNotEqual As DXMenuItem = e.Menu.Find(ClauseType.DoesNotEqual)
                    For i As Integer = e.Menu.Items.Count - 1 To 0 Step -1
                        Dim item As DXMenuItem = e.Menu.Items(i)
                        If (Not item.Equals(itemEqual)) AndAlso (Not item.Equals(itemNotEqual)) Then
                            item.Visible = False
                        End If
                    Next i
                End If
            End If
        End Sub
    End Class
End Namespace